src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -2179,10 +2179,21 @@ void CodeGen::genBMI1Intrinsic(GenTreeHWIntrinsic* node)
         case NI_BMI1_ExtractLowestSetBit:
         case NI_BMI1_GetMaskUpToLowestSetBit:
         case NI_BMI1_ResetLowestSetBit:
+        {
+            assert(op2 == nullptr);
+            assert((targetType == TYP_INT) || (targetType == TYP_LONG));
+            genHWIntrinsic_R_RM(node, ins, emitTypeSize(node->TypeGet()));
+            break;
+        }
+
         case NI_BMI1_TrailingZeroCount:
         {
             assert(op2 == nullptr);
             assert((targetType == TYP_INT) || (targetType == TYP_LONG));
+            // tzcnt has false dependency on the target register on Intel Sandy Bridge and Haswell processors,
```

Comment:
Do we want to do this unambiguously? Not all micro-architectures have this false dependency: https://reviews.llvm.org/D40334

Reply:
Adding micro-architecture checks may make troubles for CoreCLR test matrix that we have talked several times. And @damageboy's profiling shows the performance gain is really good (1.8x faster) and I use `xor r32, r32` here that is the shortest code size trade-off to fix the issue so far.